code/FromReferencesTask.cs

```diff
@@ -87,6 +101,19 @@ public override Boolean Execute()
 			}
 		}
 
+		private string GetDestinationPath(string sourceProjectBasePath, string sourceDirectoryPath)
+		{
+			Contract.Requires(sourceProjectBasePath != null);
+			Contract.Requires(sourceDirectoryPath != null);
+			Contract.Assume(sourceDirectoryPath.Length > sourceProjectBasePath.Length);
+			Contract.Assume(!String.IsNullOrEmpty(LibraryDirectoryFullPath));
+			var rootRelativePath = sourceDirectoryPath.Remove(0, sourceProjectBasePath.Length);
+			if (rootRelativePath[0] == '\\')
+			{
+				rootRelativePath = rootRelativePath.Substring(1);
+			}
+			return Path.Combine(LibraryDirectoryFullPath, rootRelativePath);
```

Comment:
Ah. Thanks for the tip. I'm not that familiar with code contracts (we don't have them in the codebase I work on) but it definitely strikes me as nicer to have the method ensuring rather than assuming after the method has returned. I suspected it was a problem with Path.Combine just not having contracts but hadn't thought to do the assume just before returning like that.